File: backend/busManagement/src/main/java/com/example/busManagement/domain/Ticket.java
package com.example.busManagement.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.util.Objects;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name="ticket")
public class Ticket {

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    @Column(name="ticket_id")
    private long id;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "person_id")
    @JsonIgnore
    private Person person;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "bus_route_id")
    @JsonIgnore
    private Bus_Route bus_route;

    @NotBlank(message = "Payment field is mandatory")
    private  String payment_method;

    @Size(min=3, max=3)
    private  String seat_number;


}

File: frontend/frontend-app/src/models/Ticket.ts
import { BusRoute } from "./BusRoute"
import { Person } from "./Person"

export interface Ticket{
    id: number,
    payment_method: string,
    seat_number: string,
    personId: number,
    bus_routeId: number,
    
    person?: Person,
    bus_route?: BusRoute,
    
    
    
    
    //noOfTicketsOfBusRouteId: number //DTO

    //"personId": 0,
    //"bus_routeId": 0,

}

